src/components/drawers/worker-drawer/worker-info/export-to-xlsx.ts

```diff
@@ -0,0 +1,313 @@
+import { VerboseDate } from "../../../../state/schedule-data/foundation-info/foundation-info.model";
+import { ShiftCode, SHIFTS } from "../../../../state/schedule-data/shifts-types/shift-types.model";
+import { ShiftHelper } from "../../../../helpers/shifts.helper";
+import { ColorHelper } from "../../../../helpers/colors/color.helper";
+
+const TOP_OFFSET = 1;
+const LEFT_OFFSET = 1;
+const NAME_ROW_WIDTH = 7;
+const NAME_ROW_HEIGHT = 2;
+const INFO_SEC_ROW_LEN = 3;
+const WEEKDAY_HEIGHT = 2;
+
+const HEIGTH = 36;
+const WIDTH = 16;
+
+export const exportToXlsx = (name: string, info, schedule: { string: string }): void => {
+  const xlsx = new WorkerCalendarXlsxExport({ name, info, schedule });
+  xlsx.formatAndSave();
+};
+
+export const WORKSHEET_NAME = "grafik";
+
+export class WorkerCalendarXlsxExport {
+  private readonly name: string;
+
+  private readonly info: { string: string };
+
+  private readonly schedule: { string: string };
+
+  constructor({ name, info, schedule }) {
+    this.name = name;
+    this.info = info;
+    this.schedule = schedule;
+  }
+
+  public formatAndSave(): void {
+    const [finalName, workbook] = this.createWorkbook();
+    FileHelper.saveToFile(workbook, finalName);
+  }
+
+  private static createWorkArea(): [xlsx.Workbook, xlsx.Worksheet] {
+    const workbook = new xlsx.Workbook();
+
+    return [
+      workbook,
+      workbook.addWorksheet(WORKSHEET_NAME, {
+        pageSetup: {
+          orientation: "portrait",
+          showGridLines: true,
+          printArea: "B2:H24",
+          fitToPage: true,
+          margins: {
+            left: 0.7,
+            right: 0.7,
+            top: 0.75,
+            bottom: 0.75,
+            header: 0.3,
+            footer: 0.3,
+          },
+          horizontalCentered: true,
+          verticalCentered: true,
+        },
+
+        properties: { defaultColWidth: 10 },
+      }),
+    ];
+  }
+
+  public createWorkbook(): [string, xlsx.Workbook] {
+    const [workbook, scheduleWorkSheet] = WorkerCalendarXlsxExport.createWorkArea();
+
+    this.setScheduleWorkSheet(scheduleWorkSheet);
+
+    const workbookName = WorkerCalendarXlsxExport.createFilename(this.name);
+    return [workbookName, workbook];
+  }
+
+  private static createFilename(name: string): string {
+    return `${name}.xlsx`;
+  }
+
+  private static createInfoSection(info: { string: number | string }): string[][] {
+    const infoSection: string[][] = [];
+
+    Object.keys(info).forEach((key) => {
+      const infoRow = [key];
+      for (let i = 0; i < INFO_SEC_ROW_LEN; i++) {
+        infoRow.push(" ");
+      }
+
+      infoRow.push(info[key]);
+      infoSection.push(infoRow);
+    });
+    return infoSection;
+  }
+
+  private static createCalendarSection(schedule): string[][] {
+    const calendar: string[][] = [];
+    let calendarDates: string[] = [];
+    let calendarShifts: string[] = [];
+
+    Object.keys(schedule).forEach((key) => {
```

Comment:
Czy `schedule` to nie jest lista?
Jeśli tak to co myślisz o tym żeby napisać `schedule.forEach((shift, idx) => ...)`?

Według mnie tak to jest mniej myłące